osquery/core/windows/process.cpp

```diff
@@ -0,0 +1,224 @@
+  
+  // In the POSIX version, the environment variable OSQUERY_WORKER is set to the string form of the
+  // child process' process ID. However, this is not easily doable on Windows. Since the value does
+  // not appear to be used by the rest of osquery, we currently just set it to '1'.
+  //
+  // For the worker case, we also set another environment variable, OSQUERY_LAUNCHER. OSQUERY_LAUNCHER
+  // stores the string form of a HANDLE to the current process. This is mostly used for detecting the 
+  // death of the launcher process in WatcherWatcherRunner::start
+  if (!setEnvVar("OSQUERY_WORKER", "1") ||
+      !setEnvVar("OSQUERY_LAUNCHER", handle.c_str())) {
+    ::CloseHandle(hLauncherProcess);
+
+    return std::shared_ptr<PlatformProcess>();
+  }
+  
+  // We don't directly use argv.c_str() as the value for lpCommandLine in CreateProcess since
+  // that argument requires a modifiable buffer. So, instead, we off-load the contents of argv
+  // into a vector which will have its backing memory as modifiable.
+  std::string argv = argv_stream.str();
+  std::vector<char> mutable_argv(argv.begin(), argv.end());
+  mutable_argv.push_back('\0');
+  
+  BOOL status = ::CreateProcessA(exec_path.c_str(),
+                                 &mutable_argv[0],
+                                 NULL,
+                                 NULL,
+                                 TRUE,
+                                 0,
+                                 NULL,
+                                 NULL,
+                                 &si,
+                                 &pi);
+  unsetEnvVar("OSQUERY_WORKER");
+  unsetEnvVar("OSQUERY_LAUNCHER");
+  ::CloseHandle(hLauncherProcess);
+  
+  if (!status) {
+    return std::shared_ptr<PlatformProcess>();
+  }
+  
+  auto process = std::make_shared<PlatformProcess>(pi.hProcess);
+  ::CloseHandle(pi.hThread);
+  ::CloseHandle(pi.hProcess);
+  
+  return process;
+}
+
+std::shared_ptr<PlatformProcess> PlatformProcess::launchExtension(const std::string& exec_path,
+                                                 const std::string& extension, 
+                                                 const std::string& extensions_socket,
+                                                 const std::string& extensions_timeout,
+                                                 const std::string& extensions_interval,
+                                                 const std::string& verbose) {
+  ::STARTUPINFOA si = { 0 };
+  ::PROCESS_INFORMATION pi = { 0 };
+  
+  si.cb = sizeof(si);
+  
+  // TODO(#1991): extension name should be sanitized or checked for invalid characters such as 
+  //              double quotes. An extension name with bad characters has the potential of affecting
+  //              command line argument parsing in the extension process which may lead to a dysfunctional
+  //              extension.
+  std::stringstream argv_stream;
+  argv_stream << "\"osquery extension: " << extension << "\" ";
```

Comment:
If it's less complicated we can add a note, not a todo, about the restriction and pick that up in another issue later on.